### DISCUSSION TITLE
Apriltag doesn't have metadata

### DISCUSSION THREAD

**Discussion by DanielChen217** [FIRST-Tech-Challenge/FtcRobotController#1099] [Q&A] [answered]
I open the prebuilt conceptapriltag on the ftc robot controller. Overall it worked but one issue i had was that the apriltag doesn't have metadata. I need the metadata since it tell us more information. As of now, my code only give the center x and y position of the tag. I have tried initiating the current library but it didn't work. i think the issue is the library not being updated but im not sure and nothing work as of now.

**Comment by texasdiaz** [comment]
One thing you didn't mention is whether or not you're using the latest software version (10.x). The metadata is embedded in the SDK, so while 9.x versions of the SDK can detect the AprilTags and provide offsets within the image, they cannot provide RBE information and related data because the SDK doesn't have any information about the tags (size, location, etc...). Update to SDK 10.x and the metadata for this season's tags is present.

**Comment by WestsideRobotics** [comment, marked as answer]
The sample OpMode "ConceptAprilTag" works correctly in Blocks and Java.  Telemetry shows the correct tag number and tag name (metadata) for all INTO THE DEEP tags 11 through 16.

I suggest you start fresh with a known build of the current season's FTC software (now at version 10.1).  Open and run the sample OpMode, without modification.

Reference info is available, starting with [this page at ftc-docs](https://ftc-docs.firstinspires.org/en/latest/apriltag/vision_portal/visionportal_overview/visionportal-overview.html).
